packages/next/src/compiled/react-server-dom-webpack-experimental/cjs/react-server-dom-webpack-server.edge.development.js

```diff
@@ -3967,6 +3967,19 @@
         bind: { value: bind, configurable: !0 }
       });
     };
+
+// This is a patch added by Next.js
+const setTimeoutOrImmediate = (
+  // edge runtime sandbox defines a stub for setImmediate
+  // (see 'addStub' in packages/next/src/server/web/sandbox/context.ts)
+  // so we can't just do this:
+  //   typeof setImmediate === 'function'
+  // luckily the stub is non-enumerable, so we can check Object.keys instead
+  Object.keys(globalThis).includes("setImmediate")
+    ? globalThis["set" + "Immediate"]
```

Comment:
Is this to trick bundlers? What happens if we access the method normally?

Reply:
Yeah it's a bundler trick, Otherwise `middleware-plugin.ts` finds it and yells:
```
A Node.js API is used (setImmediate at line: XXX) which is not supported in the Edge Runtime.
```

Reply:
We have the same issue for other APIs. This is a bug that we haven't prioritized yet. Something doesn't understand that we're not actually using this API. /cc @huozhi